addons/google_calendar/utils/google_calendar.py

```diff
@@ -85,6 +85,10 @@ def delete(self, event_id, token=None, timeout=TIMEOUT):
         url = "/calendar/v3/calendars/primary/events/%s?sendUpdates=all" % event_id
         headers = {'Content-type': 'application/json'}
         params = {'access_token': token}
+        # Delete all events from recurrence in a single request.
+        is_recurrence = self.google_service._context.get('is_recurrence', True)
+        if is_recurrence:
+            params['singleEvents'] = 'true'
```

Comment:
I don't understand it well, 
![image](https://github.com/odoo/odoo/assets/50832879/52eda48f-16bc-4a1a-9333-ee0c9790d3ef)
You delete the recurrence but singleEvents . It will do the deletion like if it was a single event. Therefore, only one mail is sent ? We get the expected results? All the recurring events are deleted ? It is strange. It feels like a hack of the API but if it works, it should definitely be documented. Don't hesitate to explain because at some point, this parameter should not work anymore. The singleEvents parameter is often used to get all the instances of a recurrence.
https://developers.google.com/calendar/api/v3/reference/events/list